From b76024d7b23bde99075d707c7a4ee568c6a3c210 Mon Sep 17 00:00:00 2001
From: Bai Ping <ping.bai@nxp.com>
Date: Mon, 16 May 2016 14:01:06 +0800
Subject: [PATCH 1063/5242] MLK-12796-02 ARM: dts: imx: Add additional pinfunc
 define for imx6ull

commit  2064d93a51481d27f29724636743bf163c30320e from
https://source.codeaurora.org/external/imx/linux-imx.git

On i.MX6ULL, the pin MUX and CTRL register of BOOT_MODEx and TAMPERx
pins have been move to the IOMUXC_SNVS. Add additional pinfunc define
and correct the pinctrl binding.

Signed-off-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/imx6ull-14x14-ddr3-arm2-cs42888.dts   |    2 +-
 .../boot/dts/imx6ull-14x14-ddr3-arm2-wm8958.dts    |    2 +-
 arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts      |   26 ++++++++++++++------
 3 files changed, 20 insertions(+), 10 deletions(-)

diff --git a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-cs42888.dts b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-cs42888.dts
index 367b0ed..5130398 100644
--- a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-cs42888.dts
+++ b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-cs42888.dts
@@ -135,7 +135,7 @@
 
 &sai2 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_sai2>;
+	pinctrl-0 = <&pinctrl_sai2 &pinctrl_sai2_hp_det_b>;
 	assigned-clocks = <&clks IMX6UL_CLK_SAI2_SEL>,
 			<&clks IMX6UL_CLK_SAI2>;
 	assigned-clock-parents = <&clks IMX6UL_CLK_PLL4_AUDIO_DIV>;
diff --git a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-wm8958.dts b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-wm8958.dts
index 0e55c83..fbbc648 100644
--- a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-wm8958.dts
+++ b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-wm8958.dts
@@ -102,7 +102,7 @@
 
 &sai2 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_sai2>;
+	pinctrl-0 = <&pinctrl_sai2 &pinctrl_sai2_hp_det_b>;
 	assigned-clocks = <&clks IMX6UL_CLK_SAI2_SEL>,
 			<&clks IMX6UL_CLK_SAI2>;
 	assigned-clock-parents = <&clks IMX6UL_CLK_PLL4_AUDIO_DIV>;
diff --git a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts
index cfbcc11..fcd1c84 100644
--- a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts
@@ -402,13 +402,6 @@
 			>;
 		};
 
-		pinctrl_bt: btgrp {
-			fsl,pins = <
-				MX6UL_PAD_SNVS_TAMPER6__GPIO5_IO06	0x80000000
-				MX6UL_PAD_SNVS_TAMPER9__GPIO5_IO09	0x80000000
-				MX6UL_PAD_SNVS_TAMPER5__GPIO5_IO05	0x80000000
-			>;
-		};
 
 		pinctrl_csi1: csi1grp {
 			fsl,pins = <
@@ -656,7 +649,6 @@
 				MX6UL_PAD_SD1_DATA2__SAI2_RX_DATA     0x110b0
 				MX6UL_PAD_SD1_DATA3__SAI2_TX_DATA     0x1f0b8
 				MX6UL_PAD_SD1_CLK__SAI2_MCLK          0x1b0b0
-				MX6UL_PAD_SNVS_TAMPER0__GPIO5_IO00   0x17059
 			>;
 		};
 
@@ -852,6 +844,24 @@
 	};
 };
 
+&iomuxc_snvs {
+	imx6ul-ddr3-arm2 {
+		pinctrl_bt: btgrp {
+			fsl,pins = <
+				MX6ULL_PAD_SNVS_TAMPER6__GPIO5_IO06	0x80000000
+				MX6ULL_PAD_SNVS_TAMPER9__GPIO5_IO09	0x80000000
+				MX6ULL_PAD_SNVS_TAMPER5__GPIO5_IO05	0x80000000
+			>;
+		};
+
+		pinctrl_sai2_hp_det_b: sai2_hp_det_grp {
+			fsl,pins = <
+				MX6ULL_PAD_SNVS_TAMPER0__GPIO5_IO00   0x17059
+			>;
+		};
+	};
+};
+
 &lcdif {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_lcdif_dat
-- 
1.7.9.5

